airflow/api_connexion/endpoints/xcom_endpoint.py

```diff
@@ -26,18 +32,35 @@ def delete_xcom_entry():
     raise NotImplementedError("Not implemented yet.")
 
 
-def get_xcom_entries():
+@provide_session
+def get_xcom_entries(dag_id, dag_run_id, task_id, session):
     """
     Get all XCom values
     """
-    raise NotImplementedError("Not implemented yet.")
+    query = session.query(XCom)
+    query = query.filter(and_(XCom.dag_id == dag_id,
+                              XCom.task_id == task_id))
+    query = query.join(DR, and_(XCom.dag_id == DR.dag_id, XCom.execution_date == DR.execution_date))
+    query = query.filter(DR.run_id == dag_run_id)
+    return xcom_collection_schema.dump(XComCollection(xcom_entries=query.all(), total_entries=query.count()))
```

Comment:
does `query.count()` issue another count query to the db? if so, we should call `len` on rows returned by `query.all()` instead.

Reply:
total_entries should contain the number of all items on all pages. This allows us to build pagination and build the user interface more easily.

Reply:
Then there should be some order-by if I'm not wrong @mik-laj . Can we look at this - https://github.com/wizeline/sqlalchemy-pagination 
What is ur take on this?